### ISSUE TITLE
False positive errors on struct pattern with raw identifiers

### ISSUE THREAD

**Issue by fujiapple852** [intellij-rust/intellij-rust#4341]
## Environment

* **IntelliJ Rust plugin version:**
0.2.104.2132-192

* **Rust toolchain version:**
rustc 1.37.0 (eae3437df 2019-08-13)

* **IDE name and version:**
CLion 2019.2.1 Build #CL-192.6262.62, built on August 21, 2019

* **Operating system:**
OSX

## Problem description
False positive errors on struct pattern with raw identifiers:

    Struct pattern does not correspond to its declaration [E0027]
    Extra field found in the struct pattern: 'r#type' [E0026]

## Steps to reproduce

```rust
struct Foo {
    r#type: u64
}

fn bar() {
    let Foo { r#type: _ } = Foo { r#type: 0 };
}
```
